src/main/java/com/cinema_booking/booking/room/RoomController.java

```diff
@@ -0,0 +1,83 @@
+package com.cinema_booking.booking.room;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.cinema_booking.booking.room.dto.RoomDTO;
+import com.cinema_booking.booking.room.model.Room;
+
+@Controller
+@RestController
+@RequestMapping("/api/room")
+public class RoomController {
+
+    private final RoomService roomService;
+
+    @Autowired
+    public RoomController(RoomService roomService) {
+        this.roomService = roomService;
+    }
+
+    @GetMapping("/list")
+    public ResponseEntity<List<RoomDTO>> getRoomList() {
+        try {
+            List<RoomDTO> roomDTOList = roomService.getRoomDTOList();
+            return new ResponseEntity<>(roomDTOList, HttpStatus.OK);
+        } catch (Exception error) {
+            return new ResponseEntity<>(null, HttpStatus.BAD_REQUEST);
```

Comment:
Idea is great, but as you see you have to catch exception in every single endpoint. Check out `@ControllerAdvice`